### ISSUE TITLE
Periodical maintenance operation for Dovecot mailbox format

### ISSUE THREAD

**Issue by gmasse** [tomav/docker-mailserver#1697]
## Is your feature request related to a problem?
Dovecot [dbox format](https://github.com/tomav/docker-mailserver#dovecot_mailbox_format) requires periodical purge to free unused space. Details in the official documentation:
> Expunging a message only decreases the message's refcount. The space is later freed in "purge" step. This is typically done in a nightly cronjob when there's less disk I/O activity. The purging first finds all files that have refcount=0 mails. Then it goes through each file and copies the refcount>0 mails to other mdbox files (to the same files as where newly saved messages would also go), updates the map index and finally deletes the original file. So there is never any overwriting or file truncation.
> 
> The purging can be invoked explicitly running doveadm purge.

## Describe the solution you'd like
Daily crontab within the container:
```
$ cat /etc/cron.d/dovecot-purge
#
#  Dovecot purge maintenance
#
# m h dom mon dow user	command
38 2	* * *	root    doveadm purge -A
```
**Question:** should this file be added automatically when enabling dbox format?
(DOVECOT_MAILBOX_FORMAT=sdbox or mdbox)

## Alternatives you've considered
Run from the host's crontab (not in the container).

## What are you going to contribute?
I can implement the solution.

**Comment by aendeavor** [comment]
Thanks for this post. As the dbox format is marked experimental, I think it is fine to add this Cron entry unconditionally. Therefore, from my point of view, the answer to

>  should this file be added automatically when enabling dbox format?

is yes.

You're very welcome with providing and implementing a solution. I will review it when the PR is ready.

PS: I dislike the idea of running Cron from the host and not inside the container. We should not consider this alternative. Nevertheless worth the mention so we can rule this out.

**Comment by erik-wramner** [comment]
@gmasse please make sure that you delete the cron job if dbox format is not used as well (if it is present of course).

**Comment by aendeavor** [comment]
This somehow wasn't linked. Closed with #1705.
